### PR TITLE
Node Documentation Todos

### PR DESCRIPTION
**Pull request by kavins14**
Updated the astroid nodes according to the notes in the `TODO.md` file. Some of the todos were already complete but the note was not removed (nodes `Compare`, `AsyncFunctionDef`, `AsyncFor`).

### CODE REVIEW

File: nodes/AugAssign.py [hunk 1]
```diff
@@ -21,10 +20,11 @@
     if necessary (as in BinOp).
 
 
-Example:
-    - target  -> [Name(id='x', ctx=Store())]
-    - value   -> Num(n=1)
-    - op      -> Add()
+Example 1:
```
Comment (david-yz-liu) [inline nodes/AugAssign.py]: You shouldn't add the numbering here unless you plan to add another example.

Comment (david-yz-liu) [reply nodes/AugAssign.py]: (The same below as well.)

File: nodes/BoolOp.py [hunk 2]
```diff
@@ -4,18 +4,34 @@
 A boolean operation, 'or' or 'and'.
 
 Attributes:
-    - values  (Expr)
-        - The values involved.
-    - op      (Or | And)
+    - values  ([Expr])
+        - A list of values involved
```
Comment (david-yz-liu) [inline nodes/BoolOp.py]: Say here "A list of the argument expressions" to be a bit more specific

File: nodes/BoolOp.py [hunk 3]
```diff
@@ -4,18 +4,34 @@
...
+    - op      (str)
         - The operator, Or or And.
```
Comment (david-yz-liu) [inline nodes/BoolOp.py]: While you're updating this, say `'or' or 'and'` (with the quotes!) because case does matter.

File: nodes/TODO.md [hunk 4]
```diff
@@ -1,63 +1,4 @@
...
-Should specifically reference other astroid nodes that use `Comprehension`, e.g. `ListComp`.
-
-Should give an example of using an `if`; also, the example seems to contradict the statement that `ifs` is None (rather than an empty list) if there are no `if`s.
-
-Finally, the variable name given in the example documentation is incorrect.
-
-## Const
-
-The description of `value` should really reference the fact that these are literal constants rather than just computed values.
+# Node
```
Comment (david-yz-liu) [inline nodes/TODO.md]: I'm not sure what you intended by what you added here, but you can just delete this whole file now that all the ToDOs have been addressed.

File: nodes/AugAssign.py (hunk 1 above)
Comment (kavins14) [reply nodes/AugAssign.py]: I have made changes accordingly.

File: nodes/BoolOp.py (hunk 2 above)
Comment (kavins14) [reply nodes/BoolOp.py]: I have made changes accordingly.

File: nodes/BoolOp.py (hunk 3 above)
Comment (kavins14) [reply nodes/BoolOp.py]: I have made changes accordingly.

File: nodes/TODO.md (hunk 4 above)
Comment (kavins14) [reply nodes/TODO.md]: I have made changes accordingly.

**Comment by kavins14** [comment]
I looked into every node documentation and fixed it so that the examples are consistent with the astroid ast representation of those examples. I also merged `upstream/master` into `origin/master` and into this branch.

**Pull request merged**